From c4faed6d0b6f30346a6ad1d89fc9067e6c4ce26f Mon Sep 17 00:00:00 2001
From: Dan Gittik <dangittik@google.com>
Date: Fri, 13 Jul 2018 15:10:39 +0100
Subject: [PATCH 33/44] Fixed auto-brightness first screen update.

When auto-brightness is enabled, it should update the screen as soon
as ALS data is available; however, if that data results in the same
screen brightness as before, this update is optimised away, even
though it should be applied (in case the actual screen brightness
did change, e.g. by a user who turned adaptive brightness off).

Test: Turn adaptive brightness off; set the screen brightness to 0;
      turn adaptive brightness on again (do so several times).
      Before the fix, it would often take a while for the screen
      to brighten (because similar ALS data results in the same
      screen brightness as before, an thus no update is issued).
      After the fix, it should brighten immediately.

Fixes: 111381286

(cherry picked from commit f387dd7188f74869f28a9c01d19cff22906c51f2)

Change-Id: If68765281587db887b81e2774c87e81547955b2b
CRs-Fixed: 2403457
---
 .../android/server/display/AutomaticBrightnessController.java    | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/core/java/com/android/server/display/AutomaticBrightnessController.java b/services/core/java/com/android/server/display/AutomaticBrightnessController.java
index faec14c9892..9fa5fe7b70e 100644
--- a/services/core/java/com/android/server/display/AutomaticBrightnessController.java
+++ b/services/core/java/com/android/server/display/AutomaticBrightnessController.java
@@ -416,6 +416,7 @@ class AutomaticBrightnessController {
         } else if (mLightSensorEnabled) {
             mLightSensorEnabled = false;
             mAmbientLuxValid = !mResetAmbientLuxAfterWarmUpConfig;
+            mScreenAutoBrightness = -1;
             mRecentLightSamples = 0;
             mAmbientLightRingBuffer.clear();
             mCurrentLightSensorRate = -1;
-- 
2.17.1

